### PR TITLE
bend: init at 0.2.9

### LINKED ISSUE #312527: Package request: bend
**Project description**

Bend is a massively parallel, high-level programming language.

**Metadata**

* homepage URL: https://github.com/HigherOrderCO/bend
* source URL: https://github.com/HigherOrderCO/bend
* license: apache 2.0
* platforms: unix, linux, darwin

---

was able to get it compiling with this:

```nix
pkgs.rustPlatform.buildRustPackage rec {
  pname = "bend";
  version = "0.2.7";

  src = pkgs.fetchFromGitHub {
    owner = "higherorderco";
    repo = pname;
    rev = version;
    hash = "sha256-yQI+p3PT1Czruc8+OZ/lZLAkBjkchkRw/OoGCYxHTDY=";
  };

  # NOTE: broken tests - something with hvm calls
  doCheck = false;

  cargoHash = "sha256-YPPBtgxqfoKnGb77m/CUYQ7+vVgL5MHhJ8I3TQ1rqK8=";

  buildInputs = [ pkgs.hvm ];

  RUSTC_BOOTSTRAP = true;

  meta = {
    description = "A massively parallel, high-level programming language";
    homepage = "https://github.com/HigherOrderCO/bend";
    license = pkgs.lib.licenses.asl20;
  };
};
```

but when i run it like `bend run tmp.bend` (tmp.bend exists) i get this:

```
Errors:
While running hvm: No such file or directory (os error 2)
```

thought the hvm build input would fix that but no

hopefully im just being stupid somehow

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by k3yss**
## Description of changes
Add support for Bend, a massively parallel, high-level programming language from https://higherorderco.com .
resolves #312527

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by RaghavSood** [comment]
Result of `nixpkgs-review pr 312972` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>bend</li>
  </ul>
</details>

File: pkgs/by-name/be/bend/package.nix
Comment (RaghavSood) [inline pkgs/by-name/be/bend/package.nix:21]:
```suggestion
    repo = "Bend";
```

#277994

File: pkgs/by-name/be/bend/package.nix
Comment (RaghavSood) [inline pkgs/by-name/be/bend/package.nix:34]:
```suggestion
      "Bend is a massively parallel, high-level programming language";
```

From [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/baf54cdc1df81f72081e35e82a17fc01bd605aa2/pkgs/README.md?plain=1#L372):

* `meta.description` must: 
  * Not end with a period (or any punctuation for that matter).

File: pkgs/by-name/be/bend/package.nix
Comment (RaghavSood) [inline pkgs/by-name/be/bend/package.nix:32]:
```suggestion
  meta = {
```
#208242

File: pkgs/by-name/be/bend/package.nix
Comment (RaghavSood) [inline pkgs/by-name/be/bend/package.nix:39]: Upstream does support darwin - is anything holding us back from having it supported here?

If it can't be worked around or fixed, please add a comment explaining the darwin blocker so that it is possible to address it in the future

**Review by RaghavSood — COMMENTED** [review]
Please run:

```
nix shell nixpkgs/master#nixfmt-rfc-style -c nixfmt pkgs/by-name/be/bend/package.nix
```

Your commits should also be limited to:

```
maintainers: add k3yss
bend: init at 0.2.9
```

Please squash everything else

**Comment by RaghavSood** [comment]
Also appears that we do already package HVM, so this should take #313053 into account once it is merged.

File: pkgs/by-name/be/bend/package.nix
Comment (jeremyschlatter) [reply pkgs/by-name/be/bend/package.nix]:
```suggestion
```

It works for me on aarch64-darwin.

File: pkgs/by-name/be/bend/package.nix
Comment (jeremyschlatter) [inline pkgs/by-name/be/bend/package.nix:35]:
```suggestion
  nativeBuildInputs = [ hvm makeWrapper ];
  postInstall = ''
    wrapProgram $out/bin/bend \
      --prefix PATH : ${lib.makeBinPath [ hvm ]}
  '';
```

Bend requires hvm at runtime. This is one way to make it available, as suggested [here](https://github.com/NixOS/nixpkgs/issues/312527#issuecomment-2118508913).

Without this, trying to run bend against a file results in this error:

```
bend run sum.bend
Errors:
While running hvm: No such file or directory (os error 2)
```

File: pkgs/by-name/be/bend/package.nix
Comment (k3yss) [reply pkgs/by-name/be/bend/package.nix]: I have updated the code to use `platforms.unix` instead. Initially I was not considering the case where a user can have a external gpu for this/

**Comment by k3yss** [comment]
> Also appears that we do already package HVM, so this should take #313053 into account once it is merged.

I will make another PR once the version is upgraded.

**Review by jeremyschlatter — COMMENTED** [review]
Tested on aarch64-darwin, looks good to me.

Bit of a nit, but probably better to reorder the commits so `maintainers: add k3yss` comes first. As it is `lib.maintainers.k3yss` not defined when it is first used in `bend: init at 0.2.9`.

**Review by jeremyschlatter — APPROVED** [review]
LGTM

**Comment by jeremyschlatter** [comment]
Could you add `Resolves #312527` to the PR description? That will cause issue #312527 to [automatically close](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) when this is merged.

**Comment by RaghavSood** [comment]
#313053 has been merged as well

**Comment by k3yss** [comment]
It should be good to go!

**Comment by Misterio77** [comment]
Result of `nixpkgs-review pr 312972` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>bend</li>
  </ul>
</details>

**Review by Misterio77 — APPROVED** [review]

**Comment by RaghavSood** [comment]
Result of `nixpkgs-review pr 312972` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>bend</li>
  </ul>
</details>

**Comment by RaghavSood** [comment]
Result of `nixpkgs-review pr 312972` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>bend</li>
  </ul>
</details>

**Review by RaghavSood — APPROVED** [review]

**Comment by RaghavSood** [comment]
I'm able to successfully evaluate all_tree.bend from #312527, LGTM.

**Comment by k3yss** [comment]
Result of `nixpkgs-review pr 312972` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>bend</li>
  </ul>
</details>

**Pull request merged**
